raiden-ts/tests/unit/epics/transport.spec.ts

```diff
@@ -1008,10 +1060,112 @@ describe('transport epic', () => {
           ].reduce(raidenReducer, state),
         );
 
-      const promise = matrixMessageReceivedUpdateRoomEpic(action$, state$).toPromise();
-
-      // then it resolves
-      await expect(promise).resolves.toEqual(matrixRoom({ roomId }, { address: partner }));
+      await expect(
+        matrixMessageReceivedUpdateRoomEpic(action$, state$).toPromise(),
+      ).resolves.toEqual(matrixRoom({ roomId }, { address: partner }));
     });
   });
+
+  test('deliveredEpic', async () => {
+    expect.assertions(5);
+
+    const roomId = `!roomId_for_partner:${matrixServer}`,
+      processed: Processed = {
+        type: MessageType.PROCESSED,
+        message_identifier: makeMessageId(),
+      },
+      message = await signMessage(partnerSigner, processed),
+      text = encodeJsonMessage(message),
+      action = messageReceived(
+        { text, message, ts: 123, userId: partnerUserId, roomId },
+        { address: partner },
+      ),
+      other: Delivered = {
+        type: MessageType.DELIVERED,
+        delivered_message_identifier: makeMessageId(),
+      },
+      otherSigned = await signMessage(partnerSigner, other),
+      otherText = encodeJsonMessage(otherSigned),
+      otherAction = messageReceived(
+        { text: otherText, message: otherSigned, ts: 456, userId: partnerUserId, roomId },
+        { address: partner },
+      ),
+      // twice messageReceived
+      action$ = of(action, action, otherAction);
+
+    const signerSpy = jest.spyOn(depsMock.signer, 'signMessage');
+
+    const output = await deliveredEpic(action$, EMPTY, depsMock)
+      .pipe(toArray())
+      .toPromise();
+
+    expect(output).toHaveLength(2);
+    expect(output[0]).toEqual(
+      messageSend(
+        {
+          message: {
+            type: MessageType.DELIVERED,
+            delivered_message_identifier: message.message_identifier,
+            signature: expect.any(String),
+          },
+        },
+        action.meta,
+      ),
+    );
+    expect(output[1].payload.message).toBe(output[0].payload.message);
+
+    // second signature should have been cached
+    expect(signerSpy).toHaveBeenCalledTimes(1);
+    signerSpy.mockRestore();
+
+    // if you pass something different than the accepted messages, no Delivered is replied
+    await expect(
+      deliveredEpic(of(otherAction), EMPTY, depsMock).toPromise(),
+    ).resolves.toBeUndefined();
+  });
+
+  test('matrixMessageGlobalSendEpic', async () => {
+    expect.assertions(5);
+
+    const processed: Processed = {
```

Comment:
Maybe move these to fixture generator

Reply:
Quite often this `Processed` refers to the `message_identifier` of the message its confirming, not sure if it's worth the hassle to change some cases not others.

Reply:
Ok, the ones which had custom `message_identifier` were only on the `transfers.spec.ts` suite. All the others, `Processed` messages were used quite simply as a generic, easily composable `Message`, and therefore I created a `processed` fixture to cleanup these constants. Thank you for pointing this out.